compiler/rustc_resolve/src/lib.rs

```diff
@@ -881,6 +881,16 @@ impl<'a> NameBindingData<'a> {
             invoc_parent_expansion.is_descendant_of(self_parent_expansion);
         !(certainly_before_other_or_simultaneously || certainly_before_invoc_or_simultaneously)
     }
+
+    fn determined(&self) -> bool {
+        match &self.kind {
+            NameBindingKind::Import { binding, import, .. } => {
+                import.parent_scope.module.unexpanded_invocations.borrow().is_empty()
```

Comment:
Could you add an assert ensuring that this import is a glob import?

Reply:
The only non-recursive use of this function is in `fn resolve_ident_in_module_unadjusted_ext` where it's a glob import.

Reply:
The second use is in `fn determined` itself, and it that case it doesn't seem to work correctly.

For single imports it doesn't matter whether `unexpanded_invocations` is empty or not, for them if the binding is set once it cannot be overridden later by an expanded macro.
If the function is called for non-glob imports, then it should return `true` for them without checking `unexpanded_invocations`.